foucaultview.cpp

```diff
@@ -152,6 +169,328 @@ QVector<QPoint> scaleProfile(QPolygonF points, int width,
+            painter.setPen(QPen(QColor(Qt::yellow), 3));
+            painter.drawLines(profilePoints);
+        }
+        painter.restore();
+        label->setPixmap(pix);
+    };
+
+    // 4. Update the actual labels
+    paintAndDisplay(ui->ronchiViewLb, ronchiImg, settings.ronchiX * settings.rocOffset);
+    paintAndDisplay(ui->foucaultViewLb, foucaultImg, settings.rocOffset);
+
+    QApplication::restoreOverrideCursor();
+}
+
+
+
+void foucaultView::generateBatchRonchiImage(const QList<wavefront*>& wavefrontList)
+{
+    // 1. Initial Checks
+    if (wavefrontList.isEmpty() || !m_wf) return;
+
+    // 2. Ask user for Grid Layout
+    bool ok;
+    int cols = QInputDialog::getInt(this, tr("Batch Ronchi"),
+                                    tr("Number of columns:"), 2, 1, 10, 1, &ok);
+    if (!ok) return;
+
+    // 3. Prepare Optical Settings from UI
+    OpticalTestSettings s;
+    s.rocOffset      = ui->rocOffsetSb->value();
+    s.ronchiX        = ui->RonchiX->value();
+    s.lpi            = ui->lpiSb->value();
+    s.gamma          = ui->gammaSb->value();
+    s.slitWidth      = ui->slitWidthSb->value();
+    s.useMM          = ui->useMM->isChecked();
+    s.movingSource   = ui->movingSourceRb->isChecked();
+    s.knifeOnLeft    = ui->knifeOnLeftCb->isChecked();
+    s.clearCenter    = ui->clearCenterCb->isChecked();
+    s.heightMultiply = this->heightMultiply;
+    s.lateralOffset  = this->lateralOffset;
+    s.outputLambda   = 550.0;
+
+    // 4. Calculate Grid and Canvas Geometry
+    int count = wavefrontList.size();
+    int rows = (count + cols - 1) / cols;
+    int imgDim = m_wf->data.cols;
+
+    int headerHeight = 70; // Space for the parameters at the top
+    int textBuffer = 40;   // Space for labels at the bottom of each image
+
+    int cellW = imgDim;
+    int cellH = imgDim + textBuffer;
+
+    // Total Canvas size
+    QImage canvas(cellW * cols, (cellH * rows) + headerHeight, QImage::Format_RGB32);
+    canvas.fill(Qt::black);
+
+    QPainter painter(&canvas);
+    painter.setRenderHint(QPainter::Antialiasing);
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+
+    // 5. Draw Simulation Header (Removed Slit)
+    painter.setPen(Qt::white);
+    painter.setFont(QFont("Arial", 12, QFont::Bold));
+    QString unit = s.useMM ? "mm" : "in";
+    QString headerText = QString("Ronchi Analysis | LPI: %1 | Offset: %2 %3 | Gamma: %4")
+                         .arg(s.lpi).arg(s.rocOffset).arg(unit).arg(s.gamma);
+    painter.drawText(20, 35, headerText);
+
+    // Draw separator line
+    painter.setPen(QPen(Qt::gray, 2));
+    painter.drawLine(10, headerHeight - 15, canvas.width() - 10, headerHeight - 15);
+
+    // 6. Loop through and Render Ronchi Patterns
+    painter.setFont(QFont("Arial", 11, QFont::Bold));
+    QFontMetrics fm(painter.font());
+
+    for (int i = 0; i < count; ++i) {
+        wavefront* currentWf = wavefrontList[i];
```

Comment:
```suggestion
for (wavefront* currentWf : wavefrontList) {
```
"range for" are realy good for such cases. No count, better readability